units/provisioning_setup_host.go

```diff
@@ -907,28 +910,47 @@ func mountLinuxVolume(ctx context.Context, env evergreen.Environment, h *host.Ho
 	}
 
 	cmd = client.CreateCommand(ctx).Sudo(true)
+
 	// skip formatting if the volume already contains a filesystem
-	cmdOut := util.NewMBCappedWriter()
-	if err = client.CreateCommand(ctx).Sudo(true).Append(fmt.Sprintf("file -s /dev/%s", deviceName)).SetOutputWriter(cmdOut).Run(ctx); err != nil {
+	output, err = h.RunJasperProcess(ctx, env, &options.Create{
+		Args: []string{"sudo", "file", "-s", deviceName},
+	})
+	if err != nil {
 		return errors.Wrap(err, "problem checking for formatted device")
 	}
-	if strings.Contains(cmdOut.String(), fmt.Sprintf("/dev/%s: data", deviceName)) {
-		cmd.Append(fmt.Sprintf("%s /dev/%s", h.Distro.HomeVolumeSettings.FormatCommand, deviceName))
+	if util.StringSliceContains(output, fmt.Sprintf("%s: data", deviceName)) {
+		cmd.Append(fmt.Sprintf("%s %s", h.Distro.HomeVolumeSettings.FormatCommand, deviceName))
 	}
 
 	cmd.Append(fmt.Sprintf("mkdir -p /%s", evergreen.HomeVolumeDir))
-	cmd.Append(fmt.Sprintf("mount /dev/%s /%s", deviceName, evergreen.HomeVolumeDir))
+	cmd.Append(fmt.Sprintf("mount %s /%s", deviceName, evergreen.HomeVolumeDir))
 	cmd.Append(fmt.Sprintf("chown -R %s:%s /%s", h.User, h.User, evergreen.HomeVolumeDir))
 	cmd.Append(fmt.Sprintf("ln -s /%s %s/%s", evergreen.HomeVolumeDir, h.Distro.HomeDir(), evergreen.HomeVolumeDir))
 	if err = cmd.Run(ctx); err != nil {
 		return errors.Wrap(err, "problem running mount commands")
 	}
 
 	// write to fstab so the volume is mounted on restart
-	err = client.CreateCommand(ctx).Sudo(true).SetInputBytes([]byte(fmt.Sprintf("/dev/%s /%s auto noatime 0 0", deviceName, evergreen.HomeVolumeDir))).Append("tee --append /etc/fstab").Run(ctx)
+	err = client.CreateCommand(ctx).Sudo(true).SetInputBytes([]byte(fmt.Sprintf("%s /%s auto noatime 0 0", deviceName, evergreen.HomeVolumeDir))).Append("tee --append /etc/fstab").Run(ctx)
 	if err != nil {
 		return errors.Wrap(err, "problem appending to fstab")
 	}
 
 	return nil
 }
+
+func getMostRecentlyAddedDevice(lsblkOutput []string) (string, error) {
+	// lsblk contains at least the header and one device
+	if len(lsblkOutput) < 2 {
+		return "", errors.Errorf("lsblk output is length '%d'", len(lsblkOutput))
+	}
+
+	device := lsblkOutput[len(lsblkOutput)-1]
+	deviceNameRegexp := regexp.MustCompile(`^(?P<deviceName>sd[a-z]\d{0,2}|xvd[a-z]|hd[a-z]\d{0,2}|nvme\d{1,2}n1)`)
```

Comment:
we shouldn't use mustcompiles in regexp instances created inside of instances, because they'll panic when thefunction is called (must-compiles are better in package-level variables or inits, or things like that.) or things that are going to get called closer to startup.